### PR TITLE
Use Kokkos after its CMake overhaul

### PR DESCRIPTION
**Pull request by dalg24**
@junghans I left out Travis CI for now

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,7 @@ option(CMAKE_VERBOSE_MAKEFILE "Generate verbose Makefiles" OFF)
 include(GNUInstallDirs)
 
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
-find_package(KOKKOS REQUIRED)
+find_package(Kokkos REQUIRED)
```
Comment (junghans) [inline CMakeLists.txt]: Is the imported target still called `Kokkos::kokkos`?

Comment (dalg24) [reply CMakeLists.txt]: Yessir

**Comment by junghans** [comment]
Can you modify https://github.com/ECP-copa/Cabana/blob/master/.travis.yml#L39 to test travis as well?

File: .travis.yml [hunk 2]
```diff
@@ -88,6 +87,7 @@ script:
   - export FFLAGS="-Wall -Wextra -pedantic ${WERROR:+-Werror}" 
   - mkdir build && pushd build &&
     cmake -DCMAKE_PREFIX_PATH="$HOME/kokkos;$HOME/scafacos"
+          -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
```
Comment (junghans) [inline .travis.yml]: I don't think that is needed as we use symlinked compilers.

Comment (dalg24) [reply .travis.yml]: I removed the symlink

Comment (junghans) [reply .travis.yml]: Ok, this makes it more obvious.

**Comment by sfogerty** [comment]
Let's remember to change the wiki build instructions when this goes through.

Building Kokkos' using its develop branch using CMake

**Comment by junghans** [comment]
Hmm, I have no idea by AMPI segfaults with 4 tasks, @evan-charmworks any idea?

**Comment by matthiasdiener** [comment]
@junghans: The segfault is actually an unhandled C++ exception in the application code. Charm++/AMPI do not use exceptions, but we have a default exception handler that aborts on all uncaught exceptions: https://github.com/UIUC-PPL/charm/commit/becc7731feef74b4001fbe31a85658d054411fb6

**Comment by junghans** [comment]
@dalg24 now we are back in kokkos space for the exception.

**Comment by dalg24** [comment]
@matthiasdiener Thanks for looking into it.
I installed Charm++ this morning and wasn't able to reproduce the error.  Do you remember what line was throwing?
It really puzzles me because there is no AMPI specific code on Cabana as far as I can tell and no exception is thrown with other MPI implementations.

**Comment by matthiasdiener** [comment]
@dalg24: I haven't build this myself, but this is what I understood from the build logs in Travis. I don't know which of the lines in `Kokkos::SharedAllocationRecord::decrement` actually throws the exceptions (I think there are 3 lines in that function where exceptions could be thrown). Do you have a script to build Kokkos/Cabana?

**Comment by sslattery** [comment]
Retest this once #171 is merged

**Comment by sslattery** [comment]
If we want to depend on a Kokkos release then the merge of this PR will be blocked by the release of Kokkos 3.0

**Comment by junghans** [comment]
@dalg24 are we ready to pull the trigger here?

File: scripts/jenkins/Dockerfile [hunk 3]
```diff
@@ -35,21 +35,22 @@ RUN export OPENMPI_VERSION=4.0.0 && \
 
 # Install Kokkos
 ENV KOKKOS_DIR=/opt/kokkos
-RUN export KOKKOS_VERSION=2.9.00 && \
-    export KOKKOS_URL=https://github.com/kokkos/kokkos/archive/${KOKKOS_VERSION}.tar.gz && \
-    export KOKKOS_ARCHIVE=kokkos-${KOKKOS_VERSION}.tar.gz && \
-    export KOKKOS_SOURCE_DIR=kokkos-${KOKKOS_VERSION} && \
+RUN KOKKOS_VERSION=master && \
```
Comment (sslattery) [inline scripts/jenkins/Dockerfile]: should we fix to 3.0?

Comment (junghans) [reply scripts/jenkins/Dockerfile]: There is no master tag ;-)

Comment (junghans) [reply scripts/jenkins/Dockerfile]: Neither will there is a 3.0 one.

Comment (dalg24) [reply scripts/jenkins/Dockerfile]: You can use the revision umber if you want to be explicit and replace with a tag later.

**Pull request merged**